openshift_scalability/scripts/ingress_autoscaling/install-ingress-autoscaling.sh

```diff
@@ -0,0 +1,85 @@
+## 1. Operators -> OperatorHub
+## 2. Input keda in the filter and select 'Custom Metrics Autoscaler'.
+## 3. Click Install, in the opened page, click Install.
+## 4. Operators ->'Installed Operators', find 'Custom Metrics Autoscaler'.
+## 5. On 'All instances' tab, click "Create new" and select 'KedaController', click 'Create'.
+################################################
+set -e
+
+if [[ $(uname -a) =~ "Darwin" ]]; then 
+    os=mac
+else
+    os=linux
+fi
+
+if [[ -z $1 ]]; then
+    echo "[ERROR] please provide MAXREPLICACOUNT as parameter, e.g. 1,2,omit"
+    exit 1
+fi
+keda_pod=$(oc get po -n openshift-keda | grep Running -c)
+if [[ $keda_pod -ne 3 ]]; then
+    echo "[ERROR] please make sure KedaController is successfully installed."
+    exit 1
+fi
+echo "[INFO ]====Installing Ingresss Autoscaling===="
+
+echo "[INFO] Creating project openshift-ingress-operator"
+oc project openshift-ingress-operator
+
+echo "[INFO] Applying ConfigMap cluster-monitoring-config"
+oc apply -f content/cluster-monitoring-config.yaml
+
+echo "[INFO] Creating serviceaccount thanos"
+oc create serviceaccount thanos && oc describe serviceaccount thanos
+
+echo "[INFO ]Getting thanos-token secret"
+secret=$(oc get secret | grep thanos-token | head -n 1 | awk '{ print $1 }')
+
+echo "[INFO] Appling template keda-trigger-auth-prometheus"
+if [[ $os == "linux" ]]; then
+    sed -i s/'<token>'/$secret/g content/keda-trigger-auth-prometheus.yaml
+elif [[ $os == "mac" ]]; then
+    sed -i "" s/'<token>'/$secret/g content/keda-trigger-auth-prometheus.yaml
+fi
+oc process -f content/keda-trigger-auth-prometheus.yaml | oc apply -f -
+
+echo "[INFO] Applying thanos-metrics-reader role"
+oc apply -f content/thanos-metrics-reader.yaml
+
+echo "[INFO] Ading role thanos-metrics-reader to user thanos"
+oc adm policy add-role-to-user thanos-metrics-reader -z thanos --role-namespace=openshift-ingress-operator
+
+echo "[INFO] Ading cluster role cluster-monitoring-view to user thanos"
+oc adm policy -n openshift-ingress-operator add-cluster-role-to-user cluster-monitoring-view -z thanos
+
+echo "[INFO] Applying ScaledObject ingress-scaler"
+if [[ $1 == "omit" ]]; then
+    if [[ $os == "linux" ]]; then
+        sed -i /maxReplicaCount/d content/ingress-scaler.yaml
+    elif [[ $os == "mac" ]]; then
+        sed -i "" /maxReplicaCount/d content/ingress-scaler.yaml
+    fi
+else
+    if [[ $os == "linux" ]]; then
+        sed -i s/'<max_replica_count>'/$1/ content/ingress-scaler.yaml 
+    elif [[ $os == "mac" ]]; then
+        sed -i "" s/'<max_replica_count>'/$1/ content/ingress-scaler.yaml 
+    fi
+fi
+oc apply -f content/ingress-scaler.yaml
+
+echo "[INFO] ====Post Install Check===="
+echo "[INFO] Getting hpa"
+oc get hpa
+echo "[INFO] Getting ingresscontroller replicas"
+oc get ingresscontroller/default -o yaml -n openshift-ingress-operator | grep replicas:
+echo "[INFO] Getting number of worker nodes:"
+oc get nodes --no-headers -l node-role.kubernetes.io/worker= | wc -l
```

Comment:
I think there should be some validation here? I'm not 100% sure what should be the result. I was able to properly run with the help of the comments and the validation you added above :)

Reply:
This is a script to help enabling the feature before running the actual test cases. I haven't automate the test cases yet. But you remind me it's also good to add part of the validations here so I don't need to do them again on the test case side.